src/Task/Git/BranchName.php

```diff
@@ -95,17 +96,24 @@ public function run(ContextInterface $context): TaskResultInterface
 
             if (preg_match((string)$regex, $name)) {
                 $errors[] = sprintf('Matched blacklist rule: %s', $rule);
+                $isBlacklisted = true;
             }
         }
         foreach ($config['whitelist'] as $rule) {
             $regex = new Regex($rule);
 
             $additionalModifiersArray = array_filter(str_split($config['additional_modifiers']));
             array_map([$regex, 'addPatternModifier'], $additionalModifiersArray);
-
-            if (!preg_match((string) $regex, $name)) {
-                $errors[] = sprintf('Whitelist rule not matched: %s', $rule);
+            
+            if (preg_match((string) $regex, $name)) {
+                if ($isBlacklisted) {
```

Comment:
Can you add some tests that show what happens if there is a mix of whitelist and blacklist and how it resolves a conflict in between these regexes?

Reply:
I've added a test with a conflict and additional output.  
Does that cover what you are thinking?